### PR TITLE
Apply stacked images to position all modules

### PR DESCRIPTION
**Pull request by antarcticrainforest**
We noticed that the ```position_all_modules methods``` can "only" handle one pulse at a time. When testing the application of AGIPD geometry with karaboFai during last weeks experiment commissioning we figured that we should be able to deal with data that comes in multiple pulses. 

Hence I gave it a go and changed the ```positions_all_modules methods``` in both classes. While the method for the 'snapped' geometry was straight forward ```positions_all_modules``` of the ```AGIPD_1MGeometry``` was somewhat more tricky. 

First I tried to construct a 3D rotation matrix for a rotation around a homogeneous axis. But that did not work, it might well be that I did something wrong here, I am also not sure if that is what we should do in this case (3D rotation). This [book](https://books.google.de/books?id=J9b_CH-NrycC&lpg=PP1&pg=PA67#v=twopage&q&f=false) shows an example where they loop over the 3rd (color) axis (page 67). So I assume we also have to loop, which makes the routine even slower. [Apparently openCV is able to do an affine transformation on a  3D image](https://stackoverflow.com/questions/44674129/how-can-i-use-scipys-affine-transform-to-do-an-arbitrary-affine-transformation). But adding openCV as a requirement for only this feature is sure overkill. 

@takluyver @tmichela @zhujun98 @ebadkamil any thoughts?

### CODE REVIEW

**Comment by antarcticrainforest** [comment]
Ok, I've only edited the module positioning for the snapped geometry, should the notebook also be updated?

File: karabo_data/geometry2.py [hunk 1]
```diff
@@ -346,23 +346,22 @@ def position_all_modules(self, data):
...
                 # Offset by centre to make all coordinates positive
                 y, x = tile.corner_idx + centre
                 h, w = tile.pixel_dims
-
-                out[y:y+h, x:x+w] = tile.transform(tile_data)
+                out[..., y:y+h, x:x+w] = tile.transform(tile_data)
 
         return out, centre
 
-    def _plotting_dimensions(self):
```
Comment (takluyver) [inline karabo_data/geometry2.py]: I think we should rename this method - plotting implies 2D to me.

**Comment by takluyver** [comment]
Probably easiest not to update the notebook, because I think you'll get a merge conflict on that.

File: karabo_data/geometry2.py [hunk 2]
```diff
@@ -378,9 +377,9 @@ def _plotting_dimensions(self):
         min_yx = corners.min(axis=0) - 20
```
Comment (antarcticrainforest) [inline karabo_data/geometry2.py]: Do we want to keep these margins?

File: karabo_data/geometry2.py (hunk 1 above)
Comment (antarcticrainforest) [reply karabo_data/geometry2.py]: ok, that makes sense

File: karabo_data/geometry2.py (hunk 2 above)
Comment (takluyver) [reply karabo_data/geometry2.py]: Possibly not. I think some visual margin is useful for displaying the data, but making the array bigger than necessary is probably not the right way to achieve that.

**Pull request merged**